azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_provider.py

```diff
@@ -34,6 +34,7 @@ def __init__(self, client_id, hostname, username, ca_cert=None):
         self.on_mqtt_disconnected = None
         self.on_mqtt_published = None
         self.on_mqtt_subscribed = None
+        self.on_mqtt_unsubscribed = None
```

Comment:
maybe not for now but these are "mini-states" so I wonder if we need to track this at this level or in the transport state machine and if these can get out of sync.
Also, there will be multiple topics to publish and subscribe to, so while for now it's OK, this is going to have to change and we should start thinking about it now.

Reply:
created on board

---
In reply to: [258251940](https://github.com/Azure/azure-iot-sdk-python-preview/pull/50#discussion_r258251940) [](ancestors = 258251940)